src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,445 @@
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class Duke {
+    private static final String TASKS_FILENAME = "./data/duke.txt";
+
+    private static final String INPUT_PROMPT = "$ ";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_BYE = "bye";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_DELETE = "delete";
+
+    private static final String TASK_DEADLINE_SPLITTER = "/by";
+    private static final String TASK_EVENT_SPLITTER = "/at";
+
+    private static final String MESSAGE_WALL = "------------------------------------------------------------";
+    private static final String MESSAGE_LIST_EMPTY = "Task list is empty";
+    private static final String MESSAGE_LIST_HEADER = "Task List:";
+    private static final String MESSAGE_INVALID_TASK_NUMBER = "Invalid task number";
+    private static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
+    private static final String MESSAGE_WELCOME = "Hello! I'm Duke\nWhat can I do for you?";
+    private static final String MESSAGE_BYE = "Bye. Hope to see you again soon!";
+    private static final String MESSAGE_INVALID_TASK_TYPE = "Invalid task type";
+    private static final String MESSAGE_FILEWRITER_ERROR = "FileWriter Error";
+
+    private static final String MESSAGE_FORMAT_DONE_USAGE = "Usage: %s <task number>";
+    private static final String MESSAGE_FORMAT_TODO_USAGE = "Usage: %s <description>";
+    private static final String MESSAGE_FORMAT_DEADLINE_USAGE = "Usage: %s <description> %s <date/time>";
+    private static final String MESSAGE_FORMAT_EVENT_USAGE = "Usage: %s <description> %s <date/time>";
+    private static final String MESSAGE_FORMAT_DELETE_USAGE = "Usage: %s <task number>";
+
+    private static final String MESSAGE_FORMAT_TASK_ALREADY_MARKED = "Task #%d is already marked as done.";
+    private static final String MESSAGE_FORMAT_TASK_MARKED = "Task marked as done:\n  %s";
+    private static final String MESSAGE_FORMAT_TASK_ADDED = "Got it. Task added:\n  %s\nThere are %d tasks in the list.";
+    private static final String MESSAGE_FORMAT_EXCEPTION = "An exception has occurred.\n%s";
+    private static final String MESSAGE_FORMAT_TASK_DELETED = "Task deleted:\n  %s\nThere are %d tasks left in the list.";
+    private static final String MESSAGE_FORMAT_CREATE_FILE_FAILED = "Fail to create file - %s";
+    private static final String MESSAGE_FORMAT_FILE_NOT_FOUND = "File not found - %s";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+    private static final ArrayList<Task> TASKS = new ArrayList<>();
```

Comment:
Since this list grows and shrinks in size as tasks are added and removed from it, it is better to name the variable in lower caps as it isn't really an immutable list.